Add support for dynamically provisioning volumes

detiber opened the pull request:
Really rough first pass at adding support for dynamically provisioned volumes on top of the v1.2.0 tag.

Resolves https://github.com/openshift/origin-metrics/issues/137

detiber commented:
I'm assuming the test failure is related to my changes being based off of the v1.2.0 tag

jcantrill commented:
@detiber I'm not certain why travis is failing but its not relevant (and I dont know how to disable it)
[test]

jcantrill commented on deployer/templates/hawkular-cassandra-node-dynamic-pv.yaml:
```diff
@@ -0,0 +1,114 @@
+id: hawkular-cassandra-node-dynamic-pv
+kind: Template
+apiVersion: v1
+name: 'Hawkular Cassandra Node Template: Persistent Storage'
```
I see we are following the pattern here, but @mwringe wouldnt we rather utilize a shorter name that does not need to be enclosed in quotes when we reference it?

mwringe replied:
Where do we have to reference this by name and not by id?

jcantrill replied:
I'm missing something then.  I thought all resources had a 'name' and not an 'id'

detiber replied:
The use of id threw me off a bit as well, I've always used auto-generated ids, and set metadata.name for kube/openshift objects. I left it there just because I figured there was a reason for it.

jcantrill commented on deployer/templates/hawkular-cassandra-node-dynamic-pv.yaml:
```diff
@@ -0,0 +1,114 @@
+    labels:
+      metrics-infra: hawkular-cassandra
+    annotations:
+      volume.alpha.kubernetes.io/storage-class: "foo"
```
If we dont have a real value other then 'foo', why do we have the annotation.

detiber replied:
It's the presence of the annotation itself that causes the dynamic provisioning to happen, I used "foo" because that is what our example docs have. I'm not sure if it is possible to specify it without a value or using an empty string.

jcantrill replied:
how about 'dynamic' or does it have some other connotation like the amazon instances sizes 'c4.large'

detiber replied:
dynamic sounds good to me, will update shortly.

jcantrill commented:
[test]

detiber commented:
@jcantrill not directly related to this PR, but I'm hitting an issue with my testing where ansible is passing 'True' rather than 'true' to the deployer template, which is causing the booleans to fail.

Would it be possible to get better support for different "boolean" values?

jcantrill commented:
@detiber can you please open a separate issue that we can track and fix

detiber commented:
I reverted the rework around existing pv support. The deployer is failing validation with the dynamic provisioning, attempting to look into that issue now.

sosiouxme commented on deployer/templates/hawkular-cassandra-node-dynamic-pv.yaml:
```diff
@@ -0,0 +1,114 @@
+    labels:
+      metrics-infra: hawkular-cassandra
+    annotations:
+      volume.alpha.kubernetes.io/storage-class: "dynamic"
```
@detiber is there any harm in just setting this on created PVCs unconditionally, whether dynamic provisioning is available or not?

detiber replied:
@sosiouxme I believe so, I'm not even sure the pvc could bind to a normal volume if that annotation is set. That said, there is the possibility that a user with an environment with dynamic provisioning available would still want to use a normal PV for metrics/logging.

detiber commented:
@mwringe Current deployment error using this PR:
```
======== ERROR =========
validate_deployment_artifacts: 
ERROR: Pod hawkular-cassandra-1-wkkzi from ReplicationController hawkular-cassandra-1 could not be scheduled (placed on a node).
This is most often due to a faulty nodeSelector or lack of node availability.
There was an event for this pod with the following message:
PersistentVolumeClaim is not bound: "metrics-cassandra-1"
  * * * * 
Pod heapster-cs9wx from ReplicationController heapster is in a Pending state.
This is most often due to waiting for the container image to pull and should eventually resolve.
  * * * * 
========================
--- validate_deployed_project ---
```

The replication controller validation is mistaking the pod being in a Pending state waiting on the PVC binding as a failure.

sosiouxme commented:
Well that *is* a failure. You want to know if your pods won't actually come
up don't you...

On Fri, Jun 10, 2016 at 2:58 PM, Jason DeTiberus <notifications@github.com>
wrote:

> @mwringe <https://github.com/mwringe> Current deployment error using this
> PR:
>
> ======== ERROR =========
> validate_deployment_artifacts:
> ERROR: Pod hawkular-cassandra-1-wkkzi from ReplicationController hawkular-cassandra-1 could not be scheduled (placed on a node).
> This is most often due to a faulty nodeSelector or lack of node availability.
> There was an event for this pod with the following message:
> PersistentVolumeClaim is not bound: "metrics-cassandra-1"
>   * * * *
> Pod heapster-cs9wx from ReplicationController heapster is in a Pending state.
> This is most often due to waiting for the container image to pull and should eventually resolve.
>   * * * *
> ========================
> --- validate_deployed_project ---
>
> The replication controller validation is mistaking the pod being in a
> Pending state waiting on the PVC binding as a failure.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin-metrics/pull/138#issuecomment-225267315>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABz-u7MgagiUD3leUh54hsFAjs8fSqRks5qKbPNgaJpZM4IuiOl>
> .
>

detiber commented:
@sosiouxme 
> Well that *is* a failure. You want to know if your pods won't actually come
up don't you...

Well, it is a condition that should be checked for and and retried. With a dynamically provisioned pvc, the pods will always enter that state (assuming an initial deployment and not an update to the deployment) prior to the backing PV being created and the PVC being bound to that PV. So, in the case of a dynamically created PVC, this should be something that we retry for some determined amount of time before declaring failure.  As it is, this is considered a failure if it is seen at all and the deployer exists while the deployment happily finishes in the background.

sosiouxme commented:
That'll teach me to shoot from the hip. I didn't even read the first error
message in the context of auto-provisioning.

I agree, the unbound PVC should be a retry error, not fatal. That should
actually be easy to arrange.

On Fri, Jun 10, 2016 at 3:12 PM, Jason DeTiberus <notifications@github.com>
wrote:

> @sosiouxme <https://github.com/sosiouxme>
>
> Well that *is* a failure. You want to know if your pods won't actually
> come
> up don't you...
>
> Well, it is a condition that should be checked for and and retried. With a
> dynamically provisioned pvc, the pods will always enter that state
> (assuming an initial deployment and not an update to the deployment) prior
> to the backing PV being created and the PVC being bound to that PV. So, in
> the case of a dynamically created PVC, this should be something that we
> retry for some determined amount of time before declaring failure. As it
> is, this is considered a failure if it is seen at all and the deployer
> exists while the deployment happily finishes in the background.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin-metrics/pull/138#issuecomment-225270693>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABz-oZTx-s5yT3x8EFHBvkZfRWWmDjkks5qKbcwgaJpZM4IuiOl>
> .
>

detiber commented:
@sosiouxme I'm still getting an error even after pulling in your validation changes. I believe the problem is that event entry for the Pending pod is still being picked up. 

```
[openshift@ip-172-18-1-195 ~]$ oc logs metrics-deployer-inwxz -n openshift-infra
+ deployer_mode=deploy
+ image_prefix=detiber/origin-
+ image_version=v1.2.0-dynamic
+ master_url=https://kubernetes.default.svc:443
+ redeploy=false
+ '[' false == true ']'
+ mode=deploy
+ '[' deploy = redeploy ']'
+ cassandra_nodes=1
+ use_persistent_storage=true
+ dynamically_provision_storage=true
+ cassandra_pv_size=100Gi
+ metric_duration=7
+ heapster_node_id=nodename
+ metric_resolution=10s
+ project=openshift-infra
+ master_ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+ token_file=/var/run/secrets/kubernetes.io/serviceaccount/token
+ dir=/etc/deploy/_output
+ secret_dir=/secret
+ rm -rf /etc/deploy/_output
+ mkdir -p /etc/deploy/_output
+ chmod 700 /etc/deploy/_output
+ mkdir -p /secret
+ chmod 700 /secret
chmod: changing permissions of '/secret': Read-only file system
+ :
+ hawkular_metrics_hostname=metrics-test.redhatkeynote.com
+ hawkular_metrics_alias=hawkular-metrics
+ hawkular_cassandra_alias=hawkular-cassandra
++ date +%s
+ openshift admin ca create-signer-cert --key=/etc/deploy/_output/ca.key --cert=/etc/deploy/_output/ca.crt --serial=/etc/deploy/_output/ca.serial.txt --name=metrics-signer@1465893450
+ '[' -n 1 ']'
+ oc config set-cluster master --api-version=v1 --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt --server=https://kubernetes.default.svc:443
cluster "master" set.
++ cat /var/run/secrets/kubernetes.io/serviceaccount/token
+ oc config set-credentials account --token=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJvcGVuc2hpZnQtaW5mcmEiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIjoibWV0cmljcy1kZXBsb3llci10b2tlbi11Y2R5NSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJtZXRyaWNzLWRlcGxveWVyIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQudWlkIjoiZmE3ZTI1Y2MtMzIwYS0xMWU2LWE4OWYtMGE4ZTBkOTRmMDAxIiwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50Om9wZW5zaGlmdC1pbmZyYTptZXRyaWNzLWRlcGxveWVyIn0.vtsO7RZaw7ikViW3jFjBG6BIDtJIBPSuYMOrZtpI153JWunI7_mc-CkSpLLfKMxtB8ANQPfdIkT-_AtRyOuGT3PvLH2oYr1_TY2b15lSs_wNZVNpFX9ns52TKQuV-14GVLquWsDKNKsheLL1P1_aw4ly_T-KWRZ82XCnd_T42yD317_7h23inJoSOHf1HObeIhb0p0X_6-YVfTixua6_6mmGSVuVN9e3DEnfKCh6A1MRkcgVH0j7EceQl9rUQhleIQ9IMCPUCM8CsLX66R8hEgsY4Yq7yO-WjZlXFfP22lxOMG3IQcTHjNH1ny3kLdr8KevkmU_l-9hHKZ29TwCuVA
user "account" set.
+ oc config set-context current --cluster=master --user=account --namespace=openshift-infra
context "current" set.
+ oc config use-context current
switched to context "current".
+ old_kc=/etc/deploy/.kubeconfig
+ KUBECONFIG=/etc/deploy/_output/kube.conf
+ '[' -z 1 ']'
+ oc config set-cluster deployer-master --api-version=v1 --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt --server=https://kubernetes.default.svc:443
cluster "deployer-master" set.
++ cat /var/run/secrets/kubernetes.io/serviceaccount/token
+ oc config set-credentials deployer-account --token=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJvcGVuc2hpZnQtaW5mcmEiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIjoibWV0cmljcy1kZXBsb3llci10b2tlbi11Y2R5NSIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJtZXRyaWNzLWRlcGxveWVyIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQudWlkIjoiZmE3ZTI1Y2MtMzIwYS0xMWU2LWE4OWYtMGE4ZTBkOTRmMDAxIiwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50Om9wZW5zaGlmdC1pbmZyYTptZXRyaWNzLWRlcGxveWVyIn0.vtsO7RZaw7ikViW3jFjBG6BIDtJIBPSuYMOrZtpI153JWunI7_mc-CkSpLLfKMxtB8ANQPfdIkT-_AtRyOuGT3PvLH2oYr1_TY2b15lSs_wNZVNpFX9ns52TKQuV-14GVLquWsDKNKsheLL1P1_aw4ly_T-KWRZ82XCnd_T42yD317_7h23inJoSOHf1HObeIhb0p0X_6-YVfTixua6_6mmGSVuVN9e3DEnfKCh6A1MRkcgVH0j7EceQl9rUQhleIQ9IMCPUCM8CsLX66R8hEgsY4Yq7yO-WjZlXFfP22lxOMG3IQcTHjNH1ny3kLdr8KevkmU_l-9hHKZ29TwCuVA
user "deployer-account" set.
+ oc config set-context deployer-context --cluster=deployer-master --user=deployer-account --namespace=openshift-infra
context "deployer-context" set.
+ '[' -n 1 ']'
+ oc config use-context deployer-context
switched to context "deployer-context".
+ case $deployer_mode in
+ validate_preflight
+ set -e
+ set +x

PREFLIGHT CHECK SUCCEEDED
validate_master_accessible: ok
validate_deployer_secret:
+ setup_certificate hawkular-metrics hawkular-metrics,metrics-test.redhatkeynote.com ''
+ local name=hawkular-metrics
+ local hostnames=hawkular-metrics,metrics-test.redhatkeynote.com
+ local envVar=
+ '[' -n '' ']'
+ '[' -s /secret/hawkular-metrics.pem ']'
+ openshift admin ca create-server-cert --key=/etc/deploy/_output/hawkular-metrics.key --cert=/etc/deploy/_output/hawkular-metrics.crt --hostnames=hawkular-metrics,metrics-test.redhatkeynote.com --signer-cert=/etc/deploy/_output/ca.crt --signer-key=/etc/deploy/_output/ca.key --signer-serial=/etc/deploy/_output/ca.serial.txt
+ cat /etc/deploy/_output/hawkular-metrics.key /etc/deploy/_output/hawkular-metrics.crt
+ cp /etc/deploy/_output/ca.crt /etc/deploy/_output/hawkular-metrics-ca.cert
+ setup_certificate hawkular-cassandra hawkular-cassandra ''
+ local name=hawkular-cassandra
+ local hostnames=hawkular-cassandra
+ local envVar=
+ '[' -n '' ']'
+ '[' -s /secret/hawkular-cassandra.pem ']'
+ openshift admin ca create-server-cert --key=/etc/deploy/_output/hawkular-cassandra.key --cert=/etc/deploy/_output/hawkular-cassandra.crt --hostnames=hawkular-cassandra --signer-cert=/etc/deploy/_output/ca.crt --signer-key=/etc/deploy/_output/ca.key --signer-serial=/etc/deploy/_output/ca.serial.txt
+ cat /etc/deploy/_output/hawkular-cassandra.key /etc/deploy/_output/hawkular-cassandra.crt
+ cp /etc/deploy/_output/ca.crt /etc/deploy/_output/hawkular-cassandra-ca.cert
+ echo 'Generating randomized passwords for the Hawkular Metrics and Cassandra keystores and truststores'
Generating randomized passwords for the Hawkular Metrics and Cassandra keystores and truststores
++ cat /dev/urandom
++ tr -dc _A-Z-a-z-0-9
++ head -c15
+ hawkular_metrics_keystore_password=AZulQPYAhsRqFVW
++ cat /dev/urandom
++ tr -dc _A-Z-a-z-0-9
++ head -c15
+ hawkular_metrics_truststore_password=UsnSvKvT102VFIz
++ cat /dev/urandom
++ tr -dc _A-Z-a-z-0-9
++ head -c15
+ hawkular_cassandra_keystore_password=iM97zZXIsSWjXhx
++ cat /dev/urandom
++ tr -dc _A-Z-a-z-0-9
++ head -c15
+ hawkular_cassandra_truststore_password=TwrUoDxr9-RAWTX
+ echo 'Creating the Hawkular Metrics keystore from the PEM file'
Creating the Hawkular Metrics keystore from the PEM file
+ openssl pkcs12 -export -in /etc/deploy/_output/hawkular-metrics.pem -out /etc/deploy/_output/hawkular-metrics.pkcs12 -name hawkular-metrics -noiter -nomaciter -password pass:AZulQPYAhsRqFVW
+ keytool -v -importkeystore -srckeystore /etc/deploy/_output/hawkular-metrics.pkcs12 -srcstoretype PKCS12 -destkeystore /etc/deploy/_output/hawkular-metrics.keystore -deststoretype JKS -deststorepass AZulQPYAhsRqFVW -srcstorepass AZulQPYAhsRqFVW
Entry for alias hawkular-metrics successfully imported.
Import command completed:  1 entries successfully imported, 0 entries failed or cancelled
[Storing /etc/deploy/_output/hawkular-metrics.keystore]
+ echo 'Creating the Hawkular Cassandra keystore from the PEM file'
+ openssl pkcs12 -export -in /etc/deploy/_output/hawkular-cassandra.pem -out /etc/deploy/_output/hawkular-cassandra.pkcs12 -name hawkular-cassandra -noiter -nomaciter -password pass:iM97zZXIsSWjXhx
Creating the Hawkular Cassandra keystore from the PEM file
+ keytool -v -importkeystore -srckeystore /etc/deploy/_output/hawkular-cassandra.pkcs12 -srcstoretype PKCS12 -destkeystore /etc/deploy/_output/hawkular-cassandra.keystore -deststoretype JKS -deststorepass iM97zZXIsSWjXhx -srcstorepass iM97zZXIsSWjXhx
Entry for alias hawkular-cassandra successfully imported.
Import command completed:  1 entries successfully imported, 0 entries failed or cancelled
[Storing /etc/deploy/_output/hawkular-cassandra.keystore]
+ echo 'Creating the Hawkular Metrics Certificate'
+ keytool -noprompt -export -alias hawkular-metrics -file /etc/deploy/_output/hawkular-metrics.cert -keystore /etc/deploy/_output/hawkular-metrics.keystore -storepass AZulQPYAhsRqFVW
Creating the Hawkular Metrics Certificate
Certificate stored in file </etc/deploy/_output/hawkular-metrics.cert>
+ echo 'Creating the Hawkular Cassandra Certificate'
Creating the Hawkular Cassandra Certificate
+ keytool -noprompt -export -alias hawkular-cassandra -file /etc/deploy/_output/hawkular-cassandra.cert -keystore /etc/deploy/_output/hawkular-cassandra.keystore -storepass iM97zZXIsSWjXhx
Certificate stored in file </etc/deploy/_output/hawkular-cassandra.cert>
+ echo 'Importing the Hawkular Metrics Certificate into the Cassandra Truststore'
+ keytool -noprompt -import -v -trustcacerts -alias hawkular-metrics -file /etc/deploy/_output/hawkular-metrics.cert -keystore /etc/deploy/_output/hawkular-cassandra.truststore -trustcacerts -storepass TwrUoDxr9-RAWTX
Importing the Hawkular Metrics Certificate into the Cassandra Truststore
Certificate was added to keystore
[Storing /etc/deploy/_output/hawkular-cassandra.truststore]
+ echo 'Importing the Hawkular Cassandra Certificate into the Hawkular Metrics Truststore'
Importing the Hawkular Cassandra Certificate into the Hawkular Metrics Truststore
+ keytool -noprompt -import -v -trustcacerts -alias hawkular-cassandra -file /etc/deploy/_output/hawkular-cassandra.cert -keystore /etc/deploy/_output/hawkular-metrics.truststore -trustcacerts -storepass UsnSvKvT102VFIz
Certificate was added to keystore
[Storing /etc/deploy/_output/hawkular-metrics.truststore]
+ echo 'Importing the Hawkular Cassandra Certificate into the Cassandra Truststore'
Importing the Hawkular Cassandra Certificate into the Cassandra Truststore
+ keytool -noprompt -import -v -trustcacerts -alias hawkular-cassandra -file /etc/deploy/_output/hawkular-cassandra.cert -keystore /etc/deploy/_output/hawkular-cassandra.truststore -trustcacerts -storepass TwrUoDxr9-RAWTX
Certificate was added to keystore
[Storing /etc/deploy/_output/hawkular-cassandra.truststore]
+ echo 'Importing the CA Certificate into the Cassandra Truststore'
+ keytool -noprompt -import -v -trustcacerts -alias ca -file /etc/deploy/_output/ca.crt -keystore /etc/deploy/_output/hawkular-cassandra.truststore -trustcacerts -storepass TwrUoDxr9-RAWTX
Importing the CA Certificate into the Cassandra Truststore
Certificate was added to keystore
[Storing /etc/deploy/_output/hawkular-cassandra.truststore]
+ keytool -noprompt -import -v -trustcacerts -alias metricca -file /etc/deploy/_output/hawkular-metrics-ca.cert -keystore /etc/deploy/_output/hawkular-cassandra.truststore -trustcacerts -storepass TwrUoDxr9-RAWTX
Certificate was added to keystore
[Storing /etc/deploy/_output/hawkular-cassandra.truststore]
+ keytool -noprompt -import -v -trustcacerts -alias cassandraca -file /etc/deploy/_output/hawkular-cassandra-ca.cert -keystore /etc/deploy/_output/hawkular-cassandra.truststore -trustcacerts -storepass TwrUoDxr9-RAWTX
Certificate was added to keystore
[Storing /etc/deploy/_output/hawkular-cassandra.truststore]
+ echo 'Importing the CA Certificate into the Hawkular Metrics Truststore'
Importing the CA Certificate into the Hawkular Metrics Truststore
+ keytool -noprompt -import -v -trustcacerts -alias ca -file /etc/deploy/_output/ca.crt -keystore /etc/deploy/_output/hawkular-metrics.truststore -trustcacerts -storepass UsnSvKvT102VFIz
Certificate was added to keystore
[Storing /etc/deploy/_output/hawkular-metrics.truststore]
+ keytool -noprompt -import -v -trustcacerts -alias metricsca -file /etc/deploy/_output/hawkular-metrics-ca.cert -keystore /etc/deploy/_output/hawkular-metrics.truststore -trustcacerts -storepass UsnSvKvT102VFIz
Certificate was added to keystore
[Storing /etc/deploy/_output/hawkular-metrics.truststore]
+ keytool -noprompt -import -v -trustcacerts -alias cassandraca -file /etc/deploy/_output/hawkular-cassandra-ca.cert -keystore /etc/deploy/_output/hawkular-metrics.truststore -trustcacerts -storepass UsnSvKvT102VFIz
Certificate was added to keystore
[Storing /etc/deploy/_output/hawkular-metrics.truststore]
++ cat /dev/urandom
++ tr -dc _A-Z-a-z-0-9
++ head -c15
+ hawkular_metrics_password=tb1nHKPw-28gu3X
+ htpasswd -cb /etc/deploy/_output/hawkular-metrics.htpasswd hawkular tb1nHKPw-28gu3X
Adding password for user hawkular
+ echo

Creating the Hawkular Metrics Secrets configuration json file
+ echo 'Creating the Hawkular Metrics Secrets configuration json file'
+ cat
++ base64 -w 0 /etc/deploy/_output/hawkular-metrics.keystore
++ base64
+++ echo AZulQPYAhsRqFVW
++ base64 -w 0 /etc/deploy/_output/hawkular-metrics.truststore
++ base64
+++ echo UsnSvKvT102VFIz
++ base64
+++ echo hawkular-metrics
++ base64 -w 0 /etc/deploy/_output/hawkular-metrics.htpasswd
+ echo
+ echo 'Creating the Hawkular Metrics Certificate Secrets configuration json file'
+ cat

Creating the Hawkular Metrics Certificate Secrets configuration json file
++ base64 -w 0 /etc/deploy/_output/hawkular-metrics.cert
++ base64 -w 0 /etc/deploy/_output/hawkular-metrics-ca.cert
+ echo
+ echo 'Creating the Hawkular Metrics User Account Secrets'
+ cat

Creating the Hawkular Metrics User Account Secrets
++ base64
+++ echo hawkular
++ base64
+++ echo tb1nHKPw-28gu3X
+ echo
+ echo 'Creating the Cassandra Secrets configuration file'
+ cat

Creating the Cassandra Secrets configuration file
++ base64 -w 0 /etc/deploy/_output/hawkular-cassandra.keystore
++ base64
+++ echo iM97zZXIsSWjXhx
++ base64
+++ echo hawkular-cassandra
++ base64 -w 0 /etc/deploy/_output/hawkular-cassandra.truststore
++ base64
+++ echo TwrUoDxr9-RAWTX
++ base64 -w 0 /etc/deploy/_output/hawkular-cassandra.pem
+ echo
+ echo 'Creating the Cassandra Certificate Secrets configuration json file'
+ cat

Creating the Cassandra Certificate Secrets configuration json file
++ base64 -w 0 /etc/deploy/_output/hawkular-cassandra.cert
++ base64 -w 0 /etc/deploy/_output/hawkular-cassandra-ca.cert
+ echo 'Creating Hawkular Metrics & Cassandra Secrets'
Creating Hawkular Metrics & Cassandra Secrets
+ oc create -f /etc/deploy/_output/hawkular-metrics-secrets.json
secret "hawkular-metrics-secrets" created
+ oc create -f /etc/deploy/_output/hawkular-metrics-certificate.json
secret "hawkular-metrics-certificate" created
+ oc create -f /etc/deploy/_output/hawkular-metrics-account.json
secret "hawkular-metrics-account" created
+ oc create -f /etc/deploy/_output/cassandra-secrets.json
secret "hawkular-cassandra-secrets" created
+ oc create -f /etc/deploy/_output/cassandra-certificate.json
secret "hawkular-cassandra-certificate" created
+ echo 'Creating Hawkular Metrics & Cassandra Templates'
+ oc create -f templates/hawkular-metrics.yaml
Creating Hawkular Metrics & Cassandra Templates
template "hawkular-metrics" created
+ oc create -f templates/hawkular-cassandra.yaml
template "hawkular-cassandra-services" created
+ oc create -f templates/hawkular-cassandra-node-pv.yaml
template "hawkular-cassandra-node-pv" created
+ oc create -f templates/hawkular-cassandra-node-dynamic-pv.yaml
template "hawkular-cassandra-node-dynamic-pv" created
+ oc create -f templates/hawkular-cassandra-node-emptydir.yaml
template "hawkular-cassandra-node-emptydir" created
+ oc create -f templates/support.yaml
template "hawkular-support" created
Deploying Hawkular Metrics & Cassandra Components
+ echo 'Deploying Hawkular Metrics & Cassandra Components'
+ oc process hawkular-metrics -v IMAGE_PREFIX=detiber/origin-,IMAGE_VERSION=v1.2.0-dynamic,METRIC_DURATION=7,MASTER_URL=https://kubernetes.default.svc:443
+ oc create -f -
service "hawkular-metrics" created
replicationcontroller "hawkular-metrics" created
+ oc process hawkular-cassandra-services
+ oc create -f -
service "hawkular-cassandra" created
service "hawkular-cassandra-nodes" created
+ oc process hawkular-support -v HAWKULAR_METRICS_HOSTNAME=metrics-test.redhatkeynote.com
+ oc create -f -
serviceaccount "hawkular" created
serviceaccount "cassandra" created
route "hawkular-metrics" created
+ '[' true = true ']'
+ '[' true = true ']'
+ echo 'Setting up Cassandra with Dynamically Provisioned Storage'
Setting up Cassandra with Dynamically Provisioned Storage
+ oc process hawkular-cassandra-node-dynamic-pv -v IMAGE_PREFIX=detiber/origin-,IMAGE_VERSION=v1.2.0-dynamic,NODE=1,PV_SIZE=100Gi,MASTER=true
+ oc create -f -
persistentvolumeclaim "metrics-cassandra-1" created
replicationcontroller "hawkular-cassandra-1" created
++ seq 2 1
+ deploy_heapster
+ set -ex
+ '[' -n '' ']'
+ '[' -s /secret/heapster.cert ']'
+ openshift admin ca create-server-cert --key=/etc/deploy/_output/heapster.key --cert=/etc/deploy/_output/heapster.cert --hostnames=heapster --signer-cert=/etc/deploy/_output/ca.crt --signer-key=/etc/deploy/_output/ca.key --signer-serial=/etc/deploy/_output/ca.serial.txt
+ '[' -n '' ']'
+ '[' -s /secret/heapster-allowed-users ']'
+ echo system:master-proxy
+ '[' -n '' ']'
+ '[' -s /secret/heapster-client-ca.cert ']'
+ cp /var/run/secrets/kubernetes.io/serviceaccount/ca.crt /etc/deploy/_output/heapster_client_ca.cert
+ echo

Creating the Heapster Secrets configuration json file
+ echo 'Creating the Heapster Secrets configuration json file'
+ cat
++ base64 -w 0 /etc/deploy/_output/heapster.cert
++ base64 -w 0 /etc/deploy/_output/heapster.key
++ base64 -w 0 /etc/deploy/_output/heapster_client_ca.cert
++ base64 -w 0 /etc/deploy/_output/heapster_allowed_users
+ echo 'Installing the Heapster Component.'
Installing the Heapster Component.
Creating the Heapster secret
+ echo 'Creating the Heapster secret'
+ oc create -f /etc/deploy/_output/heapster-secrets.json
secret "heapster-secrets" created
+ echo 'Creating the Heapster template'
+ '[' -n '' ']'
+ oc create -f templates/heapster.yaml
Creating the Heapster template
template "hawkular-heapster" created
+ echo 'Deploying the Heapster component'
+ '[' -n '' ']'
Deploying the Heapster component
+ oc process hawkular-heapster -v IMAGE_PREFIX=detiber/origin-,IMAGE_VERSION=v1.2.0-dynamic,MASTER_URL=https://kubernetes.default.svc:443,NODE_ID=nodename,METRIC_RESOLUTION=10s
+ oc create -f -
serviceaccount "heapster" created
service "heapster" created
replicationcontroller "heapster" created
+ validate_deployment
+ set -e
+ set +x
=========================
VALIDATING THE DEPLOYMENT
--- validate_nodes_accessible ---
--- validate_deployment_artifacts ---
======== ERROR =========
validate_deployment_artifacts: 
ERROR: Pod hawkular-cassandra-1-30nz2 from ReplicationController hawkular-cassandra-1 could not be scheduled (placed on a node).
This is most often due to a faulty nodeSelector or lack of node availability.
There was an event for this pod with the following message:
PersistentVolumeClaim is not bound: "metrics-cassandra-1"
  * * * * 
Pod heapster-mq9xp from ReplicationController heapster is in a Pending state.
This is most often due to waiting for the container image to pull and should eventually resolve.
  * * * * 
========================
--- validate_deployed_project ---

VALIDATION FAILED
[openshift@ip-172-18-1-195 ~]$ oc get pods -n openshift-infra
NAME                         READY     STATUS    RESTARTS   AGE
hawkular-cassandra-1-30nz2   1/1       Running   0          5m
hawkular-metrics-q6sux       1/1       Running   0          5m
heapster-mq9xp               1/1       Running   0          5m
metrics-deployer-inwxz       0/1       Error     0          7m
```

mwringe commented:
Closed at this PR was rebased and copied over as https://github.com/openshift/origin-metrics/pull/192

detiber commented:
@mwringe apologies about letting this sit like that, never got a chance to pick it back up with AnsibleFest piggybacking Summit like it did.

mwringe commented:
@detiber not a problem, thanks for doing the work for this pr :)
